lms/djangoapps/courseware/access_response.py

```diff
@@ -54,15 +57,26 @@ def to_json(self):
             "has_access": self.has_access,
             "error_code": self.error_code,
             "developer_message": self.developer_message,
-            "user_message": self.user_message
+            "user_message": self.user_message,
+            "user_fragment": self.user_fragment,
         }
 
     def __repr__(self):
-        return "AccessResponse({!r}, {!r}, {!r}, {!r})".format(
+        return "AccessResponse({!r}, {!r}, {!r}, {!r}, {!r})".format(
             self.has_access,
             self.error_code,
             self.developer_message,
-            self.user_message
+            self.user_message,
+            self.user_fragment,
+        )
+
+    def __eq__(self, other):
+        if not isinstance(other, AccessResponse):
+            return False
+
+        return (
+            self.has_access == other.has_access and
+            self.error_code == other.error_code
         )
```

Comment:
It's not necessary to compare the other fields?

Reply:
Yeah, I think I'd feel more comfortable with that. It's just a little less surprising.